src/compiler/commandLineParser.ts

```diff
@@ -2468,7 +2500,12 @@ namespace ts {
         basePath: string, errors: Push<Diagnostic>, configFileName?: string): CompilerOptions {
 
         const options = getDefaultCompilerOptions(configFileName);
-        convertOptionsFromJson(optionDeclarations, jsonOptions, basePath, options, Diagnostics.Unknown_compiler_option_0, errors);
+
+        const diagnostics = {
+            unknownOptionDiagnostic: Diagnostics.Unknown_compiler_option_0,
+            unknownDidYouMeanDiagnostic: Diagnostics.Unknown_compiler_option_0_Did_you_mean_1 ,
+        };
+        convertOptionsFromJson(optionDeclarations, jsonOptions, basePath, options, diagnostics, errors);

```

Comment:
Can you instead pass defaultDiagnostics (also would recommend naming it compilerOptionsDefaultDiagnostics )